corehq/apps/data_dictionary/views.py

```diff
@@ -424,42 +444,51 @@ def _process_bulk_upload(bulk_file, domain):
     errors = []
     import_fhir_data = toggles.FHIR_INTEGRATION.enabled(domain)
     fhir_resource_type_by_case_type = {}
-    expected_columns_in_prop_sheet = 5
```

Comment:
I haven't read over the PR, but from a glance it looks like this function is getting quite large.  What would you think about making a `bulk.py`, moving all this stuff there, and breaking it out into some helper functions?

Reply:
I'm in favour for this. It would also make testing easier if we can break this down into smaller chunks. Thanks for the suggestion, I'll see what I can do.

Reply:
I have done the refactoring ([65919d1](https://github.com/dimagi/commcare-hq/pull/33259/commits/65919d13fec2584c38b352f5e9c5ec96e4222ba2)). @esoergel Let me know if you think that this should be broken down further.

Reply:
Thanks for splitting that apart - I find that kind of thing makes it easier to work with - fewer variables to hold in your head.  I'll leave it up to you guys to decide on the specifics, nothing blocking on my end!